x-pack/solutions/observability/plugins/infra/server/routes/custom_dashboards/lib/find_custom_dashboard.ts

```diff
@@ -1,33 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import { type SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
-import { INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE } from '../../../saved_objects';
-import type {
-  InfraCustomDashboard,
-  InfraCustomDashboardAssetType,
-} from '../../../../common/custom_dashboards';
-
-export async function findCustomDashboard(
-  assetType: InfraCustomDashboardAssetType,
-  savedObjectsClient: SavedObjectsClientContract
-) {
-  const result = await savedObjectsClient.find<InfraCustomDashboard>({
-    type: INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
```

Comment:
Should we also remove [the related saved object](https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/infra/server/saved_objects/custom_dashboards/custom_dashboards_saved_object.ts) as well and maybe migrate it and remove the related data there as it's no longer needed? Or at least it can be deprecated in the newer versions only to keep it backwards compatible if we need to. It can be done in a follow-up but IMO it will be good to do it at once as it can create confusion. wdyt?